fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergAlterTableExecutor.java

```diff
@@ -495,4 +531,110 @@ private void expireSnapshots(List<ConstantOperator> args) {
+            if (snapshot.manifestListLocation() != null) {
+                validFileNames.add(fileName(snapshot.manifestListLocation()));
+            }
+
+            for (ManifestFile manifest : snapshot.allManifests(table.io())) {
+                if (!processedManifestFilePaths.add(manifest.path())) {
+                    continue;
+                }
+
+                validFileNames.add(fileName(manifest.path()));
+                try (ManifestReader<? extends ContentFile<?>> manifestReader = readerForManifest(table, manifest)) {
+                    for (ContentFile<?> contentFile : manifestReader) {
+                        validFileNames.add(fileName(contentFile.location()));
+                    }
+                } catch (IOException e) {
+                    throw new StarRocksConnectorException("Unable to list manifest file content from " + manifest.path(), e);
+                }
+            }
+        }
+
+        metadataFileLocations(table, false).stream()
+                .map(IcebergUtil::fileName)
+                .forEach(validFileNames::add);
+
+        statisticsFilesLocations(table).stream()
+                .map(IcebergUtil::fileName)
+                .forEach(validFileNames::add);
+
+        validFileNames.add("version-hint.text");
+
+        actions.add(() -> scanAndDeleteInvalidFiles(table.location(), olderThanMillis, validFileNames));
+    }
+
+    private static ManifestReader<? extends ContentFile<?>> readerForManifest(Table table, ManifestFile manifest) {
+        return switch (manifest.content()) {
+            case DATA -> ManifestFiles.read(manifest, table.io());
+            case DELETES -> ManifestFiles.readDeleteManifest(manifest, table.io(), table.specs());
+        };
+    }
+
+    private void scanAndDeleteInvalidFiles(String tableLocation, long expiration, Set<String> validFiles) {
+        try {
+            URI uri = new Path(tableLocation).toUri();
+            FileSystem fileSystem = FileSystem.get(uri, hdfsEnvironment.getConfiguration());
+            RemoteIterator<LocatedFileStatus> allFiles = fileSystem.listFiles(new Path(tableLocation), true);
+            List<Path> filesToDelete = new ArrayList<>();
+            while (allFiles.hasNext()) {
+                LocatedFileStatus entry = allFiles.next();
+                FileStatus status = fileSystem.getFileStatus(entry.getPath());
+                if (status.getModificationTime() < expiration && !validFiles.contains(entry.getPath().getName())) {
+                    filesToDelete.add(entry.getPath());
+                    if (filesToDelete.size() >= DELETE_BATCH_SIZE) {
+                        filesToDelete.forEach(path -> {
+                            try {
+                                fileSystem.delete(path, false);
+                                LOGGER.debug("Deleted file {}", path);
+                            } catch (IOException e) {
+                                LOGGER.error("Failed to delete file {}", path, e);
```

Comment:
This exceprtion would only pring error in log, and wil not catch by the outermoset try-catch